lib/msal-node/src/client/ClientCredentialClient.ts

```diff
@@ -73,14 +73,14 @@ export class ClientCredentialClient extends BaseClient {
     ): Promise<AuthenticationResult | null> {
         const cachedAccessToken = this.readAccessTokenFromCache();
 
+        // must refresh due to non-existent access_token
         if (!cachedAccessToken) {
             this.serverTelemetryManager?.setCacheOutcome(
                 CacheOutcome.NO_CACHED_ACCESS_TOKEN
             );
             return null;
-        }
-
-        if (
+        // must refresh due to the expires_in value
+        } else if (
```

Comment:
You don't need any of these `else if` because you already `return null`. Having fewer branches makes the code easier to read. Have a look at https://learn.microsoft.com/en-us/visualstudio/code-quality/code-metrics-cyclomatic-complexity?view=vs-2022